### PR TITLE
Adding base class for HxrDiffractometer

### PR DESCRIPTION
**Pull request by c-tsoi**
## Description

Adding base class for HxrDiffractometer, this class will ask the user to input the prefix HXR:GON:MMS: and it has the motion stages hardcoded.
## Motivation and Context

This change allows us to control the hxr diffractometer stages using python.
## How Has This Been Tested?

I was able to connect to the PVs and I have done motion test to the stages.
## Pre-merge checklist
- [ ] Code works interactively
- [ ] Code contains descriptive docstrings, including context and API
- [ ] New/changed functions and methods are covered in the test suite where possible
- [ ] Test suite passes locally
- [ ] Test suite passes on GitHub Actions
- [ ] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [ ] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by tangkong** [comment]
I think the lakeshore file was removed a while ago.  Your branch is a bit out of date, we should rebase or merge the master into this branch.  

Let me know if you need help with this

**Comment by ZLLentz** [comment]
I think the lakeshore was never part of pcdsdevices, there's a super old issue to add one and I can't find any point in time where we completed the task (#826). The file is new to me at least. Probably these should be split into two PRs, or if that is prohibitive we can do a 2-in-1 with some updated documentation.

**Comment by tangkong** [comment]
We sorted this out offline, Christian will be splitting this up into two PRs, and we rebased this one 👍

File: pcdsdevices/gon.py [hunk 1]
```diff
@@ -655,6 +657,28 @@ def format_status_info(self, status_info):
...
+
+    prefix_base : str
+        The EPICS base PV of the diffractometer stages
+    """
+
+    base_h = Cpt(BeckhoffAxis, 'BASE_H', kind='normal')
+    base_v = Cpt(BeckhoffAxis, 'BASE_V', kind='normal')
+    th = Cpt(BeckhoffAxis, 'CHI', kind='normal')
+    two_th = Cpt(BeckhoffAxis, 'TH', kind='normal')
+    chi = Cpt(BeckhoffAxis, '2TH', kind='normal')
```
Comment (vespos) [inline pcdsdevices/gon.py:677]: Fix variables names / PV suffix. th is mapping to CHI, etc

Let's call two_th - > tth instead

**Review by vespos — CHANGES_REQUESTED** [review]
Please fix the mapping and add the pre-release notes etc

**Review by vespos — APPROVED** [review]
LGTM!

**Review by ZLLentz — CHANGES_REQUESTED** [review]
I think this still needs pre-release notes

File: docs/source/upcoming_release_notes/1274-Adding_base_class_for_hxr_diffractometer..rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+--------
+- N/A
+
+Maintenance
+-----------
+- N/A
+
+Contributors
+------------
+- N/A
```
Comment (ZLLentz) [inline docs/source/upcoming_release_notes/1274-Adding_base_class_for_hxr_diffractometer..rst:30]: You gotta credit yourself tho, you made the thing

File: docs/source/upcoming_release_notes/1274-Adding_base_class_for_hxr_diffractometer..rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+--------
+- Adding base class for hxr diffractometer, this class needs the base prefix to be passed in e.g. HXR:GON:MMS. The stage suffixes are hardcoded.
+
+Device Updates
+--------------
+- N/A
+
+New Devices
+-----------
+- N/A
```
Comment (ZLLentz) [inline docs/source/upcoming_release_notes/1274-Adding_base_class_for_hxr_diffractometer..rst:18]: Old template + wrong sorting
```suggestion
Library Features
--------
- N/A

Device Features
--------------
- N/A

New Devices
-----------
- Adding base class for hxr diffractometer, this class needs the base prefix to be passed in e.g. HXR:GON:MMS. The stage suffixes are hardcoded.
```

File: docs/source/upcoming_release_notes/1274-Adding_base_class_for_hxr_diffractometer..rst (hunk 2 above)
Comment (ZLLentz) [inline docs/source/upcoming_release_notes/1274-Adding_base_class_for_hxr_diffractometer..rst:30]:
```suggestion
- c-tsoi
```

**Review by ZLLentz — APPROVED** [review]

**Pull request merged**